app/components/vacancy_selector_component/vacancy_selector.js

```diff
@@ -0,0 +1,125 @@
+import { Controller } from '@hotwired/stimulus';
+
+import accessibleAutocomplete from 'accessible-autocomplete';
+import 'accessible-autocomplete/dist/accessible-autocomplete.min.css';
+
+export default class extends Controller {
+  static targets = [
+    'fieldset',
+    'legend',
+    'radioBlock',
+    'radioItem',
+  ];
+
+  static selectorID = 'vacancy-selector';
+
+  #choiceData;
+
+  connect() {
+    if (this.radioItemTargets.length <= 20) {
+      return;
+    }
+
+    this.#hideExistingDOM();
+    this.#addLabelToHeading();
+    const hiddenField = this.#insertHiddenField();
+    const autocompleteSuggestions = this.#insertAutocompleteSuggestions();
+
+    this.#initializeAutocomplete(autocompleteSuggestions, hiddenField);
+  }
+
+  #hideExistingDOM() { this.radioBlockTarget.hidden = true; }
+
+  #addLabelToHeading() {
+    const label = document.createElement('label');
+    label.className = 'govuk-fieldset__heading';
+    label.htmlFor = this.constructor.selectorID;
+    label.innerText = this.legendTarget.innerText;
+
+    this.legendTarget.outerHTML = label.outerHTML;
+  }
+
+  #insertHiddenField() {
+    const hiddenField = document.createElement('input');
+    hiddenField.type = 'hidden';
+    hiddenField.name = this.radioItemTarget.getElementsByTagName('input').item(0).name;
+    this.fieldsetTarget.insertAdjacentElement('beforeend', hiddenField);
+
+    return hiddenField;
+  }
+
+  #insertAutocompleteSuggestions() {
+    const autocompleteSuggestions = document.createElement('div');
+    autocompleteSuggestions.className = 'autocomplete__suggestions govuk-body govuk-!-margin-bottom-0';
+
+    this.fieldsetTarget.insertAdjacentElement('afterbegin', autocompleteSuggestions);
+
+    return autocompleteSuggestions;
+  }
+
+  #buildChoiceData() {
+    if (!this.#choiceData) {
+      this.#choiceData = this.radioItemTargets.map((radio) => ({
+        id: radio.getElementsByTagName('input').item(0).value,
+        'job-title': radio.getElementsByClassName('job-title').item(0),
+        'job-ends-on': radio.getElementsByClassName('job-ends-on').item(0),
+        'job-organisation-name': radio.getElementsByClassName('job-organisation-name').item(0),
+      }));
+    }
+
+    return this.#choiceData;
+  }
+
+  #initializeAutocomplete(autocompleteSuggestions, hiddenField) {
+    accessibleAutocomplete({
+      element: autocompleteSuggestions,
+      id: this.constructor.selectorID,
+      source: this.constructor.onSearch(this.#buildChoiceData(), hiddenField),
+      templates: {
+        inputValue: this.constructor.inputValueTemplate,
+        suggestion: this.constructor.suggestionTemplate,
+      },
+      onConfirm: (choice) => {
+        if (choice) {
+          hiddenField.value = choice.id;
+        }
+      },
+    });
+  }
+
+  static inputValueTemplate(choice) {
+    if (choice) {
+      return choice['job-title'].innerText;
+    }
+
+    return undefined;
```

Comment:
This should return `undefined` by default anyway, or does ESlint want it to be explicit?

Reply:
Yeah the linter complains if there's no explicit return.